compiler/resolution/src/org/jetbrains/kotlin/resolve/multiplatform/ExpectedActualResolver.kt

```diff
@@ -50,7 +50,7 @@ object ExpectedActualResolver {
                     // TODO: support non-source definitions (e.g. from Java)
                     actual.source.containingFile != SourceFile.NO_SOURCE_FILE
                 }.groupBy { actual ->
-                    areCompatibleCallables(expected, actual)
+                    areCompatibleCallables(expected, actual, platformModule)
```

Comment:
Missed

Reply:
@max-kammerer 
No, this is the one that's required. In the `annotationsViaActualTypeAliasFromBinary` test case the function `findNamesakesFromModule` resolves the constructor of Anno to the constructor of Jnno.  This constructor is assigned to `actual` here.

Calling `actual.module` yields the module for the Java annotation which doesn't contain Anno and causes the test to fail.